src/main/java/net/earthcomputer/clientcommands/task/RenderDistanceScanTask.java

```diff
@@ -16,29 +18,70 @@
 import net.minecraft.world.level.chunk.LevelChunkSection;
 import net.minecraft.world.level.chunk.status.ChunkStatus;
 import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.Nullable;
 
-import java.util.Iterator;
+import java.lang.ref.WeakReference;
+import java.util.Set;
 import java.util.function.Predicate;
 
 public abstract class RenderDistanceScanTask extends SimpleTask {
     private static final long MAX_SCAN_TIME = 30_000_000L; // 30ms
+    private static final Set<Object> MUTEX_KEYS = Set.of(RenderDistanceScanTask.class);
+
+    static {
+        ClientLevelEvents.CHUNK_UPDATE.register((level, pos, oldState, newState) -> {
+            @Nullable WeakReference<RenderDistanceScanTask> currentScanTask = RenderDistanceScanTask.currentScanTask;
```

Comment:
Nit: there is no need to annotate local variables with `@Nullable`, IntelliJ is smart enough to infer it in most cases. Otherwise looks good.

Reply:
i can't access my computer rn, if you have a commit changes I’ll add it rq